SU-SOE/stanford_soe_helper
# SOE-2738: merged Caryl's changes

# READY FOR REVIEW

# Summary
- This adds two view blocks on the article pages for the info to be displayed for the collection it is apart of.
- This also changes the field_s_mag_article_collection in the Magazine article content type from "Unlimited" to "1" input value.


# Needed By (03/28/2018)
- Sprint end

# Criticality
- Client is eagerly awaiting this.

# Steps to Test
- Switch to this branch: https://github.com/SU-SOE/stanford_soe_helper/tree/7.x-2.x-collections-article-blocks
- Review code

Hey Bill, please update these steps to test for Greg and make sure they cover everything for the feature. Then, we can use these steps as a start for the Behat testing!!
Please update any changes in testing steps in this [ticket](https://stanfordits.atlassian.net/browse/SOE-2603)

## Verify feature
- Clear cache
 - Verify all Collection feature modules are in default state: `drush fd | grep collection`
## Add a new collection
- Add a new magazine collection at `node/add/stanford-collection`
- Verify the new magazine collection on the manage content page at `admin/manage/collection`
## Add articles to the collection
- Go to the manage content page for articles at `admin/manage/magazine_article`
- Click on `edit` for the articles you want to add to the new magazine collection (at least 18 articles)
- Flush cache
## Verify the collection page
- Visit your collection to verify the articles appear as expected (`email sign-up` and `load more` blocks should appear.
- Verify the `load more` button actually loads more blocks
## Verify the article page
- Click on an article in your collection
- Verify the items on the page: Top grey banner, Bottom block with 8 articles and a `see all` button
- Verify the `see all` button returns you to the collection page
- Verify that the `Magazine issue` block does _not_ appear on the page


# Affects 
- SoE (School of Engineering website)

# Associated Issues and/or People
## Related JIRA ticket(s)
https://stanfordits.atlassian.net/browse/SOE-2738
https://stanfordits.atlassian.net/browse/SOE-2739

## More Information

## Folks to notify
@minorwm, @cjwest, @sherakama 


# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
